cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/DefaultCredentialProviderService.java

```diff
@@ -61,6 +65,9 @@ public class DefaultCredentialProviderService extends AbstractIdleService
 
   @Override
   protected void startUp() throws Exception {
+    if (credentialProviders.size() <= 0) {
```

Comment:
Generally, any credential provider should log when it is getting initialised: https://github.com/cdapio/cdap/pull/15286/files#diff-2c272dda420a979bc45183649233a113eac7f700eccf375c27c92df749bb24a9R104

So I thought of adding the log in the case when there are no credential providers found.
If we add a log like `"Initialized %d credential providers."` it will be displayed only if there are no errors in initialising all the existing credential providers which might not satisfy the intent of the log statement.

I do not have a very strong opinion on this, can refractor WDYT?

Reply:
To clarify, I think we should always log the number of credential providers regardless of whether none were initialized or not. I agree with your intention, but if the downstream extensions don't log for whatever reason (an easy mistake to make), it's good to have some master log indicating that some (or no) extensions were initialized.

Reply:
Done.